superset/connectors/sqla/models.py

```diff
@@ -2278,8 +2285,7 @@ def write_shadow_dataset(  # pylint: disable=too-many-locals
             )
 
         # physical dataset
-        tables = []
-        if dataset.sql is None:
+        if not dataset.sql:
```

Comment:
Some of our example datasets have `.sql == ''`, which made them to be marked as virtual during the migration. it's not a big deal, since they will still work when we switch to the new models (in the new `Dataset` model the difference between virtual and physical is greatly reduced).